### PR TITLE
feat(nuxt): `useRequestHeader` utility

### PR DESCRIPTION
**Pull request by GalacticHypernova**
### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

The current request header retrival process of `useRequestHeaders` is a bit unintuitive in situations that require reading headers. It doesn't make a lot of sense from a developmental standpoint as more often than not you would need only a single header, mostly for "reading" operations (I myself have a couple of such use cases, for example, authorization middleware).
This PR provides that possibility to simplify and improve DX. With this a person could simply do
```ts
const cookie = useRequestHeader('cookie')
```
That being said, I would like to reiterate and clarify that this is **not** a replacement of `getRequestHeaders`. This can be used to directly read header values when dealing with authorization related middleware for example, whereas getRequestHeaders can be used to proxy the headers and use them in API calls, due to its object return.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by GalacticHypernova** [comment]
Marking this a draft to write the documentation.

**Comment by GalacticHypernova** [comment]
Should I also clarify in the docs that this is used as an abstraction of `getRequestHeaders` in specific header-related operations (like the example I gave in the docs)?

File: docs/3.api/2.composables/use-request-header.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    icon: i-simple-icons-github
+    to: https://github.com/nuxt/nuxt/blob/main/packages/nuxt/src/app/composables/ssr.ts
+    size: xs
+---
+
+You can use the built-in [`useRequestHeader`](/docs/api/composables/use-request-header) composable to access any incoming request header within your pages, components, and plugins.
+
+```ts
+// Get the cookie request header
+const cookie = useRequestHeader('cookie')
```
Comment (danielroe) [inline docs/3.api/2.composables/use-request-header.md:15]: I'd recommend using a different header as an example as we already have `useCookie`. Maybe `authorization`?

File: docs/3.api/2.composables/use-request-header.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+## Example
+
+We can use `useRequestHeader` to easily figure out if a user is authorized or not.
+
+The example below reads the `authorization` request header to find out if a person can access a restricted resource.
+
+```ts [middleware/authorized-only.ts]
+export default defineNuxtRouteMiddleware((to, from) => {
+  if(!useRequestHeader('authorization')){
```
Comment (danielroe) [inline docs/3.api/2.composables/use-request-header.md:30]:
```suggestion
  if (!useRequestHeader('authorization')) {
```

File: packages/nuxt/src/app/composables/ssr.ts [hunk 3]
```diff
@@ -17,6 +17,13 @@ export function useRequestHeaders (include?: any[]) {
   return Object.fromEntries(include.map(key => key.toLowerCase()).filter(key => headers[key]).map(key => [key, headers[key]]))
 }
 
+export function useRequestHeader(include: string) {
+  if (import.meta.client) { return '' }
+  const event = useRequestEvent()
+  const header = event ? getRequestHeader(event, include) : ''
+  return header
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/ssr.ts:24]:
```suggestion
  if (import.meta.client) { return undefined }
  const event = useRequestEvent()
  return event ? getRequestHeader(event, include) : undefined
```

**Comment by danielroe** [comment]
Sure, it would be good to point people to the upstream utility used 👍

File: packages/nuxt/src/app/composables/ssr.ts (hunk 3 above)
Comment (GalacticHypernova) [reply packages/nuxt/src/app/composables/ssr.ts:24]: Should this return undefined or an empty string?

File: docs/3.api/2.composables/use-request-header.md (hunk 1 above)
Comment (GalacticHypernova) [reply docs/3.api/2.composables/use-request-header.md]: Done, I added authorization in stead, thanks!

File: docs/3.api/2.composables/use-request-header.md (hunk 2 above)
Comment (GalacticHypernova) [reply docs/3.api/2.composables/use-request-header.md]: Thank you!

**Comment by Atinux** [comment]
Thank you! Indeed it is useful to be able to get one specific header and mention that it uses `useRequestHeaders` under the hood 👍

File: packages/nuxt/src/app/composables/ssr.ts (hunk 3 above)
Comment (Atinux) [reply packages/nuxt/src/app/composables/ssr.ts:24]: I would prefer returning an empty string for consistency.

Comment (danielroe) [reply packages/nuxt/src/app/composables/ssr.ts:24]: I think undefined is mroe consistent as `getRequestHeader` returns undefined if a header doesn't exist, as does `useRequestHeaders(['unknown-header'])['unknown-header']`.

File: docs/3.api/2.composables/use-request-header.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+You can use the built-in [`useRequestHeader`](/docs/api/composables/use-request-header) composable to access any incoming request header within your pages, components, and plugins.
+
+```ts
+// Get the authorization request header
+const authorization = useRequestHeader('authorization')
+```
+
+::callout
+In the browser, `useRequestHeader` will return an empty string.
```
Comment (DamianGlowala) [inline docs/3.api/2.composables/use-request-header.md:19]:
```suggestion
In the browser, `useRequestHeader` will return `undefined`.
```

**Review by danielroe — APPROVED** [review]

**Pull request merged**